### PR TITLE
Add templates for new issues and pull-requests

### PR DESCRIPTION
**Pull request by jrha**
The practices we have adopted for managing issues and pull-requests on GitHub have not been documented anywhere, this has the advantage of both documenting those practices and then directly exposing them to everyone.

See https://github.com/blog/2111-issue-and-pull-request-templates

### CODE REVIEW

**Comment by jrha** [comment]
If we decide to adopt this, I will open corresponding PRs for the other repositories.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Ensure the title of this pull-request starts with the component name followed by a colon, e.g.
+> ncm-example: demonstrate what titles should look like
+
+Describe the change you are making here, in particular we would like to know:
+
+* Why the change is necessary.
+* What backwards incompatibility it may introduce.
```
Comment (stdweird) [inline .github/PULL_REQUEST_TEMPLATE.md]: (and if so, add label)

Comment (jrha) [inline .github/PULL_REQUEST_TEMPLATE.md]: Can anyone add labels?

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Ensure the title of this issue starts with the component name followed by a colon, e.g.
+> ncm-example: demonstrate what titles should look like
+
+Describe the problem here.
```
Comment (ned21) [inline .github/ISSUE_TEMPLATE.md]: Describe the problem here, remembering to include:
- what version of the component you are using (rpm -q $component)
- what OS version you are using (/etc/redhat-release or similar)
- steps to reproduce
- actual result versus expected result

Or something similar? (I guess not all issues are bug reports so perhaps this is a bit too specific?)

Comment (jrha) [inline .github/ISSUE_TEMPLATE.md]: I've added some text below for this.

**Comment by ned21** [comment]
LGTM

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (jrha) [inline .github/PULL_REQUEST_TEMPLATE.md]: Just got a colleague to check, only us developers are privileged enough to set the labels.

**Pull request merged**